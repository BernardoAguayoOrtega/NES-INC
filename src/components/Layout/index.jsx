// import React
import React from 'react';
// import Header component
import Header from '../Header/index.jsx';

// import footer
import Footer from '../Footer/index.jsx';

// create and export Layout component
const Layout = ({ children }) => (
  <>
    <Header />
    {children}
    <Footer />
  </>
);

export default Layout;
