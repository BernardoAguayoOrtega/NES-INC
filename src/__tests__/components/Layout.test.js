// import react
import React from 'react';
// import shallow
import { shallow } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Layout from '../../components/Layout/index.jsx';

describe('<Layout />', () => {
  const wrapper = shallow(<Layout />);

  it('is the Layout rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there a footer component?', () => {
    expect(wrapper.find('Footer').exists()).toBe(true);
  });

  it('is there a header component?', () => {
    expect(wrapper.find('Header').exists()).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
