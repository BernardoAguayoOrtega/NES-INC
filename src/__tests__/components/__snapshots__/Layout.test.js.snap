// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> matches the snap shot 1`] = `
<Fragment>
  <Header />
  <Footer />
</Fragment>
`;
